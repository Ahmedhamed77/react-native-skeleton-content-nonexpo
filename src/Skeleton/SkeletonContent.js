import * as React from "react";
import PropTypes from "prop-types";
import {
  View,
  StyleSheet,
  Animated,
  Easing,
  ViewPropTypes
} from "react-native";
import LinearGradient from "react-native-linear-gradient";
import {
  DEFAULT_BONE_COLOR,
  DEFAULT_BORDER_RADIUS,
  DEFAULT_HIGHLIGHT_COLOR,
  DEFAULT_EASING,
  DEFAULT_DURATION,
  DEFAULT_ANIMATION_TYPE,
  DEFAULT_ANIMATION_DIRECTION
} from "./Constants";

export default class SkeletonContent extends React.Component {
  constructor(props) {
    super(props);
    this.containerStyle = this.props.containerStyle;
    this.duration = this.props.duration;
    this.easing = this.props.easing;
    this.highlightColor = this.props.highlightColor;
    this.boneColor = this.props.boneColor;
    this.animationType = this.props.animationType;
    this.animationDirection = this.props.animationDirection;

    this.animationPulse = new Animated.Value(0);
    this.animationShiver = new Animated.Value(0);
    this.interpolatedBackgroundColor = this.animationPulse.interpolate({
      inputRange: [0, 1],
      outputRange: [this.boneColor, this.highlightColor]
    });

    this.gradientStart = this.getGradientStartDirection();
    this.gradientEnd = this.getGradientEndDirection();

    this.state = {
      isLoading: this.props.isLoading,
      layout: this.props.layout
    };
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    if (
      nextProps.isLoading !== prevState.isLoading ||
      nextProps.layout !== prevState.layout
    ) {
      return { isLoading: nextProps.isLoading, layout: nextProps.layout };
    }
    return null;
  }

  componentDidMount() {
    this.playAnimation();
  }

  playAnimation = () => {
    if (this.animationType === "pulse") {
      Animated.loop(
        Animated.sequence([
          Animated.timing(this.animationPulse, {
            toValue: 1,
            duration: this.duration / 2,
            easing: this.easing,
            delay: this.duration
          }),
          Animated.timing(this.animationPulse, {
            toValue: 0,
            easing: this.easing,
            duration: this.duration / 2
          })
        ])
      ).start();
    } else {
      Animated.loop(
        Animated.timing(this.animationShiver, {
          toValue: 1,
          duration: this.duration,
          easing: this.easing
        })
      ).start();
    }
  };

  getGradientStartDirection = () => {
    let direction = {};
    if (this.animationType === "shiver") {
      if (
        this.animationDirection === "horizontalLeft" ||
        this.animationDirection === "horizontalRight" ||
        this.animationDirection === "verticalTop" ||
        this.animationDirection === "verticalDown" ||
        this.animationDirection === "diagonalDownRight"
      ) {
        direction = { x: 0, y: 0 };
      } else if (this.animationDirection === "diagonalTopLeft") {
        direction = { x: 1, y: 1 };
      } else if (this.animationDirection === "diagonalTopRight") {
        direction = { x: 0, y: 1 };
      } else if (this.animationDirection === "diagonalDownLeft") {
        direction = { x: 1, y: 0 };
      }
    }
    return direction;
  };

  getGradientEndDirection = () => {
    let direction = {};
    if (this.animationType === "shiver") {
      if (
        this.animationDirection === "horizontalLeft" ||
        this.animationDirection === "horizontalRight" ||
        this.animationDirection === "diagonalTopRight"
      ) {
        direction = { x: 1, y: 0 };
      } else if (
        this.animationDirection === "verticalTop" ||
        this.animationDirection === "verticalDown" ||
        this.animationDirection === "diagonalDownLeft"
      ) {
        direction = { x: 0, y: 1 };
      } else if (this.animationDirection === "diagonalTopLeft") {
        direction = { x: 0, y: 0 };
      } else if (this.animationDirection === "diagonalDownRight") {
        direction = { x: 1, y: 1 };
      }
    }
    return direction;
  };

  getBoneStyles = boneLayout => {
    let boneStyle = {
      width: boneLayout.width || 0,
      height: boneLayout.height || 0,
      borderRadius: boneLayout.borderRadius || DEFAULT_BORDER_RADIUS,
      ...boneLayout
    };
    if (this.animationType === "pulse") {
      boneStyle.backgroundColor = this.interpolatedBackgroundColor;
    } else {
      boneStyle.overflow = "hidden";
      boneStyle.backgroundColor = boneLayout.backgroundColor || this.boneColor;
    }
    return boneStyle;
  };

  getGradientTransform = boneLayout => {
    let transform = {};
    const interpolatedPosition = this.animationShiver.interpolate({
      inputRange: [0, 1],
      outputRange: this.getPositionRange(boneLayout)
    });
    if (
      this.animationDirection !== "verticalTop" &&
      this.animationDirection !== "verticalDown"
    ) {
      transform = { translateX: interpolatedPosition };
    } else {
      transform = { translateY: interpolatedPosition };
    }
    return transform;
  };

  getPositionRange = boneLayout => {
    let outputRange = [];
    const boneWidth = boneLayout.width || 0;
    const boneHeight = boneLayout.height || 0;

    if (
      this.animationDirection === "horizontalRight" ||
      this.animationDirection === "diagonalDownRight" ||
      this.animationDirection === "diagonalTopRight"
    ) {
      outputRange.push(-boneWidth, boneWidth);
    } else if (
      this.animationDirection === "horizontalLeft" ||
      this.animationDirection === "diagonalDownLeft" ||
      this.animationDirection === "diagonalTopLeft"
    ) {
      outputRange.push(boneWidth, -boneWidth);
    } else if (this.animationDirection === "verticalDown") {
      outputRange.push(-boneHeight, boneHeight);
    } else if (this.animationDirection === "verticalTop") {
      outputRange.push(boneHeight, -boneHeight);
    }
    return outputRange;
  };

  getStaticBone = layoutStyle => (
    <Animated.View style={this.getBoneStyles(layoutStyle)} />
  );

  getShiverBone = layoutStyle => (
    <View style={this.getBoneStyles(layoutStyle)}>
      <Animated.View
        style={[
          styles.absoluteGradient,
          {
            transform: [this.getGradientTransform(layoutStyle)]
          }
        ]}
      >
        <LinearGradient
          colors={[this.boneColor, this.highlightColor, this.boneColor]}
          start={this.gradientStart}
          end={this.gradientEnd}
          style={styles.gradientChild}
        />
      </Animated.View>
    </View>
  );

  getBones = (layout, children) => {
    if (layout.length > 0) {
      const iterator = Array.from(new Array(layout.length));
      return iterator.map((_, i) => {
        if (this.animationType === "pulse" || this.animationType === "none") {
          return this.getStaticBone(layout[i]);
        } else {
          return this.getShiverBone(layout[i]);
        }
      });
    } else {
      return React.Children.map(children, child => {
        const styling = child.props.style || {};
        if (this.animationType === "pulse" || this.animationType === "none") {
          return this.getStaticBone(styling);
        } else {
          return this.getShiverBone(styling);
        }
      });
    }
  };

  renderLayout = (isLoading, bones, children) => (isLoading ? bones : children);

  render() {
    const { isLoading, layout } = this.state;
    const { children } = this.props;
    const bones = this.getBones(layout, children);

    return (
      <View style={this.containerStyle}>
        {this.renderLayout(isLoading, bones, children)}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center",
    flex: 1
  },
  absoluteGradient: {
    position: "absolute",
    width: "100%",
    height: "100%"
  },
  gradientChild: {
    flex: 1
  }
});

SkeletonContent.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  layout: PropTypes.arrayOf(PropTypes.object),
  duration: PropTypes.number,
  containerStyle: ViewPropTypes.style,
  easing: PropTypes.oneOfType([typeof Easing]),
  animationType: PropTypes.oneOf(["none", "shiver", "pulse"]),
  animationDirection: PropTypes.oneOf([
    "horizontalLeft",
    "horizontalRight",
    "verticalTop",
    "verticalDown",
    "diagonalDownLeft",
    "diagonalDownRight",
    "diagonalTopLeft",
    "diagonalTopRight"
  ]),
  boneColor: PropTypes.string,
  highlightColor: PropTypes.string
};

SkeletonContent.defaultProps = {
  containerStyle: styles.container,
  easing: DEFAULT_EASING,
  duration: DEFAULT_DURATION,
  layout: [],
  animationType: DEFAULT_ANIMATION_TYPE,
  animationDirection: DEFAULT_ANIMATION_DIRECTION,
  isLoading: true,
  boneColor: DEFAULT_BONE_COLOR,
  highlightColor: DEFAULT_HIGHLIGHT_COLOR
};
