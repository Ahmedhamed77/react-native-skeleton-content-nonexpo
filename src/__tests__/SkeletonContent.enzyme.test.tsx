/**
 * @jest-environment jsdom
 */

import { shallow, mount } from "enzyme";
import React from "react";
import { View, Animated } from "react-native";
import LinearGradient from "react-native-linear-gradient";
import SkeletonContent from "../SkeletonContent";
import {
  ISkeletonContentProps,
  DEFAULT_ANIMATION_TYPE,
  DEFAULT_ANIMATION_DIRECTION,
  DEFAULT_BONE_COLOR,
  DEFAULT_EASING,
  DEFAULT_DURATION,
  DEFAULT_HIGHLIGHT_COLOR,
  DEFAULT_LOADING
} from "../Constants";

describe("SkeletonComponent testing", () => {
  it("should render empty alone", () => {
    const component = shallow(<SkeletonContent isLoading={true} />);
    expect(component).toMatchSnapshot();
  });

  it("should render Linear Gradient alone", () => {
    const props: { colors: string[] } = {
      colors: ["red", "blue"]
    };
    const component = shallow(<LinearGradient {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should have default Props", () => {
    const component = mount(<SkeletonContent isLoading={true} />);
    expect(component.props().containerStyle).toEqual({
      alignItems: "center",
      flex: 1,
      justifyContent: "center"
    });
    expect(component.props().easing).toEqual(DEFAULT_EASING);
    expect(component.props().layout).toEqual([]);
    expect(component.props().animationDirection).toEqual(
      DEFAULT_ANIMATION_DIRECTION
    );
    expect(component.props().animationType).toEqual(DEFAULT_ANIMATION_TYPE);
    expect(component.props().boneColor).toEqual(DEFAULT_BONE_COLOR);
    expect(component.props().duration).toEqual(DEFAULT_DURATION);
    expect(component.props().highlightColor).toEqual(DEFAULT_HIGHLIGHT_COLOR);
    expect(component.props().isLoading).toEqual(DEFAULT_LOADING);
    component.unmount();
  });

  it("should have the correct layout when loading", () => {
    const props: ISkeletonContentProps = {
      layout: [
        {
          width: 240,
          height: 100,
          marginBottom: 10
        },
        {
          width: 180,
          height: 40
        }
      ],
      isLoading: true,
      animationType: "none"
    };
    const component = mount(<SkeletonContent {...props} />);
    expect(component.children().find(Animated.View)).toHaveLength(2);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it("should render the correct bones for children", () => {
    const props: ISkeletonContentProps = {
      isLoading: true,
      animationType: "none"
    };
    const TestComponent = ({
      isLoading,
      animationType
    }: ISkeletonContentProps) => (
        <SkeletonContent isLoading={isLoading} animationType={animationType}>
          <View style={{ height: 100, width: 200 }} />
          <View style={{ height: 120, width: 20 }} />
          <View style={{ height: 80, width: 240 }} />
        </SkeletonContent>
      );
    const component = mount(<TestComponent {...props} />);
    expect(component.children().find(Animated.View)).toHaveLength(3);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it("should render shiver bones for children", () => {
    const props: ISkeletonContentProps = {
      isLoading: true
    };
    const TestComponent = ({ isLoading }: ISkeletonContentProps) => (
      <SkeletonContent isLoading={isLoading}>
        <Animated.View style={{ height: 100 }} />
        <Animated.View style={{ height: 120, width: 20 }} />
        <Animated.View style={{ width: 240 }} />
      </SkeletonContent>
    );
    const component = mount(<TestComponent {...props} />);
    expect(component.children().find(Animated.View)).toHaveLength(3);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it("should switch to correct children after finishing loading", () => {
    const props: ISkeletonContentProps = {
      layout: [
        {
          width: 240,
          height: 100,
          marginBottom: 10
        },
        {
          width: 180,
          height: 40
        }
      ],
      isLoading: true,
      animationType: "pulse"
    };
    const TestComponent = ({
      isLoading,
      layout,
      animationType
    }: ISkeletonContentProps) => (
        <SkeletonContent
          layout={layout}
          isLoading={isLoading}
          animationType={animationType}
        >
          <Animated.View style={{ height: 100, width: 200 }} />
          <Animated.View style={{ height: 120, width: 20 }} />
          <Animated.View />
        </SkeletonContent>
      );

    const component = mount(<TestComponent {...props} />);
    expect(component.children().find(Animated.View)).toHaveLength(2);
    component.setProps({ isLoading: false });
    expect(component.children().find(Animated.View)).toHaveLength(3);
    expect(
      component.containsAllMatchingElements([
        <View key="child" style={{ height: 100, width: 200 }} />,
        <View key="child" style={{ height: 120, width: 20 }} />,
        <View key="child" style={{ height: 80, width: 240 }} />
      ])
    );
    component.unmount();
  });

  it("should render loading children with no style", () => {
    const props: ISkeletonContentProps = {
      isLoading: true,
      animationType: "pulse"
    };
    const TestComponent = ({
      isLoading,
      animationType
    }: ISkeletonContentProps) => (
        <SkeletonContent isLoading={isLoading} animationType={animationType}>
          <Animated.View />
          <Animated.View />
        </SkeletonContent>
      );

    const component = mount(<TestComponent {...props} />);
    expect(component.children().find(Animated.View)).toHaveLength(2);
    expect(
      component.containsAllMatchingElements([
        <Animated.View key="child" />,
        <Animated.View key="child" />
      ])
    );
    component.unmount();
  });
  it("should have the correct gradient properties", () => {
    const props: ISkeletonContentProps = {
      layout: [
        {
          width: 240,
          height: 100,
          marginBottom: 10
        },
        {
          width: 180,
          height: 40
        }
      ],
      isLoading: true,
      animationDirection: "diagonalDownLeft"
    };
    const TestComponent = ({
      isLoading,
      layout,
      animationDirection
    }: ISkeletonContentProps) => (
        <SkeletonContent
          isLoading={isLoading}
          layout={layout}
          animationDirection={animationDirection}
        >
          <Animated.View style={{ height: 100, width: 200 }} />
        </SkeletonContent>
      );
    let component = mount(<TestComponent {...props} />);
    expect(component.find(LinearGradient)).toBeDefined();
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().start
    ).toEqual({
      x: 1,
      y: 0
    });
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().end
    ).toEqual({
      x: 0,
      y: 1
    });

    props.animationDirection = "diagonalTopLeft";
    component = mount(<TestComponent {...props} />);
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().start
    ).toEqual({
      x: 1,
      y: 1
    });
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().end
    ).toEqual({
      x: 0,
      y: 0
    });

    props.animationDirection = "diagonalTopRight";
    component = mount(<TestComponent {...props} />);
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().start
    ).toEqual({
      x: 0,
      y: 1
    });
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().end
    ).toEqual({
      x: 1,
      y: 0
    });

    props.animationDirection = "diagonalDownRight";
    component = mount(<TestComponent {...props} />);
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().start
    ).toEqual({
      x: 0,
      y: 0
    });
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().end
    ).toEqual({
      x: 1,
      y: 1
    });

    props.animationDirection = "verticalTop";
    component = mount(<TestComponent {...props} />);
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().start
    ).toEqual({
      x: 0,
      y: 0
    });
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().end
    ).toEqual({
      x: 0,
      y: 1
    });

    props.animationDirection = "verticalDown";
    component = mount(<TestComponent {...props} />);
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().start
    ).toEqual({
      x: 0,
      y: 0
    });
    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().end
    ).toEqual({
      x: 0,
      y: 1
    });

    expect(
      component
        .find(LinearGradient)
        .at(0)
        .props().colors
    ).toEqual([
      DEFAULT_BONE_COLOR,
      DEFAULT_HIGHLIGHT_COLOR,
      DEFAULT_BONE_COLOR
    ]);

    component.unmount();
  });
});
