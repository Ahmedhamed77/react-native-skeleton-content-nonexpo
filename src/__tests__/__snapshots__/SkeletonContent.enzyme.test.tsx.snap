// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonComponent testing should have the correct layout when loading 1`] = `
<SkeletonContent
  animationDirection="horizontalRight"
  animationType="none"
  boneColor="#E1E9EE"
  containerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  duration={1200}
  easing={[Function]}
  highlightColor="#F2F8FC"
  intensity={0.6}
  isLoading={true}
  layout={
    Array [
      Object {
        "height": 100,
        "marginBottom": 10,
        "width": 240,
      },
      Object {
        "height": 40,
        "width": 180,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <AnimatedComponent
        key="0"
        style={
          Object {
            "backgroundColor": "#E1E9EE",
            "borderRadius": 4,
            "height": 100,
            "marginBottom": 10,
            "overflow": "hidden",
            "width": 240,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#E1E9EE",
              "borderRadius": 4,
              "height": 100,
              "marginBottom": 10,
              "overflow": "hidden",
              "width": 240,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E1E9EE",
                "borderRadius": 4,
                "height": 100,
                "marginBottom": 10,
                "overflow": "hidden",
                "width": 240,
              }
            }
          />
        </View>
      </AnimatedComponent>
      <AnimatedComponent
        key="1"
        style={
          Object {
            "backgroundColor": "#E1E9EE",
            "borderRadius": 4,
            "height": 40,
            "overflow": "hidden",
            "width": 180,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#E1E9EE",
              "borderRadius": 4,
              "height": 40,
              "overflow": "hidden",
              "width": 180,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E1E9EE",
                "borderRadius": 4,
                "height": 40,
                "overflow": "hidden",
                "width": 180,
              }
            }
          />
        </View>
      </AnimatedComponent>
    </View>
  </View>
</SkeletonContent>
`;

exports[`SkeletonComponent testing should render Linear Gradient alone 1`] = `
<BVLinearGradient
  colors={
    Array [
      4294901760,
      4278190335,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
/>
`;

exports[`SkeletonComponent testing should render empty alone 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;

exports[`SkeletonComponent testing should render shiver bones for children 1`] = `
<TestComponent
  isLoading={true}
>
  <SkeletonContent
    animationDirection="horizontalRight"
    animationType="shiver"
    boneColor="#E1E9EE"
    containerStyle={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    duration={1200}
    easing={[Function]}
    highlightColor="#F2F8FC"
    intensity={0.6}
    isLoading={true}
    layout={Array []}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          key="0/.0"
          style={
            Object {
              "backgroundColor": "#E1E9EE",
              "borderRadius": 4,
              "height": 100,
              "overflow": "hidden",
              "width": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E1E9EE",
                "borderRadius": 4,
                "height": 100,
                "overflow": "hidden",
                "width": 0,
              }
            }
          >
            <AnimatedComponent
              style={
                Array [
                  Object {
                    "height": "100%",
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "translateX": -0,
                      },
                    ],
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": "100%",
                    "position": "absolute",
                    "transform": Array [
                      Object {
                        "translateX": -0,
                      },
                    ],
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": "100%",
                      "position": "absolute",
                      "transform": Array [
                        Object {
                          "translateX": -0,
                        },
                      ],
                      "width": "100%",
                    }
                  }
                >
                  <LinearGradient
                    colors={
                      Array [
                        "#E1E9EE",
                        "#F2F8FC",
                        "#E1E9EE",
                      ]
                    }
                    end={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    start={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <BVLinearGradient
                      colors={
                        Array [
                          4292995566,
                          4294113532,
                          4292995566,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 1,
                          "y": 0,
                        }
                      }
                      locations={null}
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <BVLinearGradient
                        colors={
                          Array [
                            4292995566,
                            4294113532,
                            4292995566,
                          ]
                        }
                        endPoint={
                          Object {
                            "x": 1,
                            "y": 0,
                          }
                        }
                        locations={null}
                        startPoint={
                          Object {
                            "x": 0,
                            "y": 0,
                          }
                        }
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      />
                    </BVLinearGradient>
                  </LinearGradient>
                </View>
              </View>
            </AnimatedComponent>
          </View>
        </View>
        <View
          key="1/.1"
          style={
            Object {
              "backgroundColor": "#E1E9EE",
              "borderRadius": 4,
              "height": 120,
              "overflow": "hidden",
              "width": 20,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E1E9EE",
                "borderRadius": 4,
                "height": 120,
                "overflow": "hidden",
                "width": 20,
              }
            }
          >
            <AnimatedComponent
              style={
                Array [
                  Object {
                    "height": "100%",
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "translateX": -20,
                      },
                    ],
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": "100%",
                    "position": "absolute",
                    "transform": Array [
                      Object {
                        "translateX": -20,
                      },
                    ],
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": "100%",
                      "position": "absolute",
                      "transform": Array [
                        Object {
                          "translateX": -20,
                        },
                      ],
                      "width": "100%",
                    }
                  }
                >
                  <LinearGradient
                    colors={
                      Array [
                        "#E1E9EE",
                        "#F2F8FC",
                        "#E1E9EE",
                      ]
                    }
                    end={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    start={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <BVLinearGradient
                      colors={
                        Array [
                          4292995566,
                          4294113532,
                          4292995566,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 1,
                          "y": 0,
                        }
                      }
                      locations={null}
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <BVLinearGradient
                        colors={
                          Array [
                            4292995566,
                            4294113532,
                            4292995566,
                          ]
                        }
                        endPoint={
                          Object {
                            "x": 1,
                            "y": 0,
                          }
                        }
                        locations={null}
                        startPoint={
                          Object {
                            "x": 0,
                            "y": 0,
                          }
                        }
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      />
                    </BVLinearGradient>
                  </LinearGradient>
                </View>
              </View>
            </AnimatedComponent>
          </View>
        </View>
        <View
          key="2/.2"
          style={
            Object {
              "backgroundColor": "#E1E9EE",
              "borderRadius": 4,
              "height": 0,
              "overflow": "hidden",
              "width": 240,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E1E9EE",
                "borderRadius": 4,
                "height": 0,
                "overflow": "hidden",
                "width": 240,
              }
            }
          >
            <AnimatedComponent
              style={
                Array [
                  Object {
                    "height": "100%",
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "translateX": -240,
                      },
                    ],
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": "100%",
                    "position": "absolute",
                    "transform": Array [
                      Object {
                        "translateX": -240,
                      },
                    ],
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": "100%",
                      "position": "absolute",
                      "transform": Array [
                        Object {
                          "translateX": -240,
                        },
                      ],
                      "width": "100%",
                    }
                  }
                >
                  <LinearGradient
                    colors={
                      Array [
                        "#E1E9EE",
                        "#F2F8FC",
                        "#E1E9EE",
                      ]
                    }
                    end={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    start={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <BVLinearGradient
                      colors={
                        Array [
                          4292995566,
                          4294113532,
                          4292995566,
                        ]
                      }
                      endPoint={
                        Object {
                          "x": 1,
                          "y": 0,
                        }
                      }
                      locations={null}
                      startPoint={
                        Object {
                          "x": 0,
                          "y": 0,
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <BVLinearGradient
                        colors={
                          Array [
                            4292995566,
                            4294113532,
                            4292995566,
                          ]
                        }
                        endPoint={
                          Object {
                            "x": 1,
                            "y": 0,
                          }
                        }
                        locations={null}
                        startPoint={
                          Object {
                            "x": 0,
                            "y": 0,
                          }
                        }
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      />
                    </BVLinearGradient>
                  </LinearGradient>
                </View>
              </View>
            </AnimatedComponent>
          </View>
        </View>
      </View>
    </View>
  </SkeletonContent>
</TestComponent>
`;

exports[`SkeletonComponent testing should render the correct bones for children 1`] = `
<TestComponent
  animationType="none"
  isLoading={true}
>
  <SkeletonContent
    animationDirection="horizontalRight"
    animationType="none"
    boneColor="#E1E9EE"
    containerStyle={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    duration={1200}
    easing={[Function]}
    highlightColor="#F2F8FC"
    intensity={0.6}
    isLoading={true}
    layout={Array []}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <AnimatedComponent
          key="0/.0"
          style={
            Object {
              "backgroundColor": "#E1E9EE",
              "borderRadius": 4,
              "height": 100,
              "overflow": "hidden",
              "width": 200,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E1E9EE",
                "borderRadius": 4,
                "height": 100,
                "overflow": "hidden",
                "width": 200,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#E1E9EE",
                  "borderRadius": 4,
                  "height": 100,
                  "overflow": "hidden",
                  "width": 200,
                }
              }
            />
          </View>
        </AnimatedComponent>
        <AnimatedComponent
          key="1/.1"
          style={
            Object {
              "backgroundColor": "#E1E9EE",
              "borderRadius": 4,
              "height": 120,
              "overflow": "hidden",
              "width": 20,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E1E9EE",
                "borderRadius": 4,
                "height": 120,
                "overflow": "hidden",
                "width": 20,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#E1E9EE",
                  "borderRadius": 4,
                  "height": 120,
                  "overflow": "hidden",
                  "width": 20,
                }
              }
            />
          </View>
        </AnimatedComponent>
        <AnimatedComponent
          key="2/.2"
          style={
            Object {
              "backgroundColor": "#E1E9EE",
              "borderRadius": 4,
              "height": 80,
              "overflow": "hidden",
              "width": 240,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#E1E9EE",
                "borderRadius": 4,
                "height": 80,
                "overflow": "hidden",
                "width": 240,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#E1E9EE",
                  "borderRadius": 4,
                  "height": 80,
                  "overflow": "hidden",
                  "width": 240,
                }
              }
            />
          </View>
        </AnimatedComponent>
      </View>
    </View>
  </SkeletonContent>
</TestComponent>
`;
